### PR TITLE
Add a new example about Boolean Operation

### PR DESCRIPTION
**Pull request by naveen521kk**
## Links to added or changed documentation pages

https://manimce--2228.org.readthedocs.build/en/2228/

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: docs/source/examples.rst [hunk 1]
```diff
@@ -93,6 +93,46 @@ Basic Concepts
...
+                width=4.0, height=5.0, fill_opacity=0.5, color=BLUE, stroke_width=10
+            ).move_to(LEFT)
+            ellipse2 = ellipse1.copy().set_color(color=RED).move_to(RIGHT)
+            bool_ops_text = MarkupText("<u>Boolean Operation</u>").next_to(ellipse1, UP * 3)
+            ellipse_group = Group(bool_ops_text, ellipse1, ellipse2).move_to(LEFT * 3)
+            self.play(FadeIn(ellipse_group))
+
+            i = Intersection(ellipse1, ellipse2, color=GREEN, fill_opacity=0.5)
+            self.play(i.animate.scale(0.25))
+            self.play(i.animate.move_to(RIGHT * 5 + UP * 2.5))
```
Comment (behackl) [inline docs/source/examples.rst:111]:
```suggestion
            self.play(i.animate.scale(0.25).move_to(RIGHT * 5 + UP * 2.5))
```

File: docs/source/examples.rst [hunk 2]
```diff
@@ -93,6 +93,46 @@ Basic Concepts
...
+            intersection_text = Text("Intersection", font_size=23).next_to(i, UP)
+            self.play(FadeIn(intersection_text))
+
+            u = Union(ellipse1, ellipse2, color=ORANGE, fill_opacity=0.5)
+            self.play(u.animate.scale(0.3))
+            union_text = Text("Union", font_size=23)
+            self.play(u.animate.next_to(i, DOWN, buff=union_text.height * 3))
```
Comment (behackl) [inline docs/source/examples.rst:118]:
```suggestion
            union_text = Text("Union", font_size=23)
            self.play(u.animate.scale(0.3).next_to(i, DOWN, buff=union_text.height * 3))
```

File: docs/source/examples.rst [hunk 3]
```diff
@@ -93,6 +93,46 @@ Basic Concepts
...
+            union_text.next_to(u, UP)
+            self.play(FadeIn(union_text))
+
+            e = Exclusion(ellipse1, ellipse2, color=YELLOW, fill_opacity=0.5)
+            self.play(e.animate.scale(0.3))
+            exclusion_text = Text("Exclusion", font_size=23)
+            self.play(e.animate.next_to(u, DOWN, buff=exclusion_text.height * 3.5))
```
Comment (behackl) [inline docs/source/examples.rst:125]:
```suggestion
            exclusion_text = Text("Exclusion", font_size=23)
            self.play(e.animate.scale(0.3).next_to(u, DOWN, buff=exclusion_text.height * 3.5))
```

File: docs/source/examples.rst [hunk 4]
```diff
@@ -93,6 +93,46 @@ Basic Concepts
...
+            exclusion_text.next_to(e, UP)
+            self.play(FadeIn(exclusion_text))
+
+            d = Difference(ellipse1, ellipse2, color=PINK, fill_opacity=0.5)
+            self.play(d.animate.scale(0.3))
+            difference_text = Text("Difference", font_size=23)
+            self.play(d.animate.next_to(u, LEFT, buff=difference_text.height * 3.5))
```
Comment (behackl) [inline docs/source/examples.rst:132]:
```suggestion
            difference_text = Text("Difference", font_size=23)
            self.play(d.animate.scale(0.3).next_to(u, LEFT, buff=difference_text.height * 3.5))
```

**Review by behackl — APPROVED** [review]
@naveen521kk, in case you are not heavily opposed: could we merge scaling and moving the shapes into one play call? This example currently adds about 4-5 secs of render time to the documentation build, and we could cut that down a bit if we would use fewer play calls.

(I also looked at the modified animation locally; personally I still think it looks good.)

**Comment by naveen521kk** [comment]
> could we merge scaling and moving the shapes into one play call?

I didn't understand what you meant exactly, maybe if many people like it would be good(I haven't tried it locally)?

**Comment by behackl** [comment]
> > could we merge scaling and moving the shapes into one play call?
> 
> I didn't understand what you meant exactly, maybe if many people like it would be good(I haven't tried it locally)?

I meant that you should consider applying my suggestions -- all done now. :-)

**Pull request merged**